FHIR/sushi
# Integrate FHIR Package Loader v2

**Description:** This PR integrates the brand new FHIR Package Loader (v2). The general approach was:
* Update `FHIRDefinitions` to extend `BasePackageLoader`
* Update fishing functions in `FHIRDefinitions` to use the `findResource` and `findResourceInfo` functions in `BasePackageLoader` -- ensuring that `FHIRDefinitions` still conforms to the expected `Fisher` interface.
* Remove most other functions from `FHIRDefinitions` in favor of the functions inherited from `BasePackageLoader`
* Update all affected code as necessary (most significant changes in package loading and IG Exporter)
* Update tests to load most fixtures using FPL virtual packages

This also updates the regression framework to support pre-caching of dependencies so the first run of SUSHI does not have the extra work of downloading all dependencies (which could skew results). It also reports out more information regarding performance.

**Testing Instructions:** Some ideas:
* Run it against some of your favorite projects
* Delete or rename your FHIR cache and try it on a project or two
* Try package-related features like `sushi update-dependencies`
* Try the FSH-to-FHIR API
* Run a regression.
  * Note that the regression is expected to result in changes for a few repos (mainly some differences in warning/error counts). Confirm that the changes appear to be correct and/or harmless.
